python/ray/data/_internal/logical/rules/set_read_parallelism.py

```diff
@@ -96,7 +106,9 @@ def _split_read_op_if_needed(self, op: PhysicalOperator, logical_op: Read):
             op.actual_target_max_block_size,
             op._additional_split_factor,
         )
+
         if logical_op._parallelism == -1:
+            logical_op._detected_parallelism = detected_parallelism
```

Comment:
would be better to define a method for adjusting parallelism, instead of using this attribute.